content/technotes/2018-12-13-pdftools-20.md

```diff
@@ -0,0 +1,91 @@
+---
+slug: pdftools-20
+title: "Pdftools 2.0: powerful pdf text extraction tools"
+date: '2018-12-13'
+authors:
+  - name: Jeroen Ooms
+    twitter: opencpu
+categories: technotes
+cover: https://cdn.flipsnack.com/blog/wp-content/uploads/2017/10/09164009/page-turning-pdf.jpg
+tags:
+  - R
+  - pdf
+  - images
+  - ocr
+---
+
+A new version of pdftools has been released to CRAN. Go get it while it's hot:
+
+```r
+install.packages("pdftools")
+```
+
+This version has two major improvements: low level text extraction and encoding improvements.
+
+## Low level text extraction
+
+A pdf document may seem to contain paragraphs or tables in a viewer, but this is not actually true. Pdf is a _printing format_. A pdf page consists of many unrelated lines, bitmaps, and textboxes with a given size, position and content. Hence a table in a pdf file is really just a large unordered, nicely positioned set of lines and words. This makes sense for printing, but makes extracting text or data from a pdf file extremely difficult. 
+
+Because the pdf has little semantic structure, the `pdf_text()` function in pdftools has to render the PDF to a text canvas, in order to find sentences or paragraphs. It does so pretty well, but some users have asked for something more low level.
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr"><a href="https://twitter.com/opencpu?ref_src=twsrc%5Etfw">@opencpu</a> any chance pdftext could return a data frame with one row for each text box with columns giving page, positions, and contents?</p>&mdash; Hadley Wickham (@hadleywickham) <a href="https://twitter.com/hadleywickham/status/945729495956369409?ref_src=twsrc%5Etfw">December 26, 2017</a></blockquote>
```

Comment:
why not use the Hugo shortcode? https://gohugo.io/content-management/shortcodes/#example-tweet-input